lib/internal/test_runner/runner.js

```diff
@@ -446,21 +457,42 @@ function run(options) {
   }
 
   const root = createTestTree({ concurrency, timeout, signal });
-  const testFiles = files ?? createTestFileList();
+  let testFiles = files ?? createTestFileList();
 
   let postRun = () => root.postRun();
   let filesWatcher;
   if (watch) {
+    // Overriding testFiles so we won't try to consume again the stream
+    testFiles = ArrayIsArray(testFiles) ? testFiles : testFiles.toArray();
+
+    // TODO - should not pass a promise
```

Comment:
Any idea how can I avoid passing promise? as the tests need to have the `filesWatcher`, should I just mark the `run` function as `async` and  await it normally?

Reply:
`testFiles` should probably just be async iterable